### PR TITLE
Merge all vec256 tests into one framework

### PR DESCRIPTION
**Pull request by glaringlee**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#47294 Merge all vec256 tests into one framework**

We are aiming to merge all the vec256 tests into the framework the created in `vec256_test_all_types.h/cpp`.
Currently, the vec256 tests in `vec256_test.cpp` are not 100% covered by this framework. (80% covered).

This PR is to enhence vec256 test framework in the to support all the tests that we currently not supported from `vec256_test.cpp`

Progress:
- [x] Frac
- [x] Fmod
- [x] Blendv (For non-complex only since blendv for complex is not supported in vec256 base at the moment)
- [x] copy/set
- [x] arange
- [x] convert_check
- [x] fmadd 

Test coverage: (Manually run the oss coverage on ATen/cpu/vec256 folder under gcc9 using gcov)

Before change: (vec256_test.cpp and vec256_test_all_types.cpp together)
Lines:        326 Hit, 985 Total, 33.1% Coverage
Functions: 105 Hit, 288 Total, 36.5% Coverage
 
After change: (vec256_test_all_types.cpp only)
Lines:        318 Hit, 981 Total, 32.4% Coverage
Functions: 109 Hit, 289 Total, 37.7% Coverage

We basically covered all the tests from vec256_test.cpp

Performance:
By merging these tests, the total testing time reduced around 10% on CI machines.

Differential Revision: [D24707378](https://our.internmc.facebook.com/intern/diff/D24707378)

### CODE REVIEW

File: aten/src/ATen/test/CMakeLists.txt [hunk 1]
```diff
@@ -90,7 +90,6 @@ list(APPEND ATen_MOBILE_TEST_SRCS
   ${CMAKE_CURRENT_SOURCE_DIR}/quantized_test.cpp)
 
 list(APPEND ATen_VEC256_TEST_SRCS
-  ${CMAKE_CURRENT_SOURCE_DIR}/vec256_test.cpp
```
Comment (VitalyFedyunin) [inline aten/src/ATen/test/CMakeLists.txt:93]: You might need to delete this file and .h file.

**Review by VitalyFedyunin — COMMENTED** [review]
Overall looks good, with couple comments:
1) It is better to delete unlinked files.
2) Have you tried to look test coverage diff

**Comment by glaringlee** [comment]
> Overall looks good, with couple comments:
> 
> 1. It is better to delete unlinked files.
> 2. Have you tried to look test coverage diff

1. Will do.
2. Haven't noticed that the coverage test stopped running on CI, I will manually run this locally and post the result.

**Comment by glaringlee** [comment]
Updated test coverage info in PR description.

**Comment by glaringlee** [comment]
Sync'ed with @malfet . This is good to go, ARM test can be done later.

**Review by VitalyFedyunin — APPROVED** [review]

**Pull request closed without merging**